docs/source/en/guides/upload.md

```diff
@@ -103,6 +103,70 @@ set, files are uploaded at the root of the repo.
+
+---------- 2024-07-22 17:23:17 (0:00:00) ----------
+Files:   hashed 5/5 (5.0G/5.0G) | pre-uploaded: 0/5 (0.0/5.0G) | committed: 0/5 (0.0/5.0G) | ignored: 0
+Workers: hashing: 0 | get upload mode: 0 | pre-uploading: 5 | committing: 0 | waiting: 11
+---------------------------------------------------
+```
+
+First, the repo is created if it didn't exist before. Then the local folder is scanned for files to upload. For each file, we try to recover metadata information (from a previously interrupted upload). From there, it is able to launch workers and print an update status every 1 minute. Here we can see that 5 files have already been hashed but not pre-uploaded. 5 workers are pre-uploading files while the 11 others are waiting for a task.
+
+A command line is also provided. You can define the number of workers and the level of verbosity in the terminal:
+
+```sh
+huggingface-cli large-upload HuggingFaceM4/Docmatix --repo-type=dataset /path/to/local/docmatix --num-workers=16
+```
+
+<Tip>
+
+For large uploads, you have to set `repo_type="model"` or `--repo-type=model` explicitly. Usually this information is implicit in all other `HfApi` methods. This is to avoid having data uploaded to a repository with a wrong type. If that's the case, you'll have to re-upload everything.
+
+</Tip>
+
+### Tips and tricks for large uploads
+
+There are some limitations to be aware of when dealing with a large amount of data in your repo. Given the time it takes to stream the data, getting an upload/push to fail at the end of the process or encountering a degraded experience, be it on hf.co or when working locally, can be very annoying.
+
+Check out our [Repository limitations and recommendations](https://huggingface.co/docs/hub/repositories-recommendations) guide for best practices on how to structure your repositories on the Hub. Next, let's move on with some practical tips to make your upload process as smooth as possible.
+
+- **Start small**: We recommend starting with a small amount of data to test your upload script. It's easier to iterate on a script when failing takes only a little time.
+- **Expect failures**: Streaming large amounts of data is challenging. You don't know what can happen, but it's always best to consider that something will fail at least once -no matter if it's due to your machine, your connection, or our servers. For example, if you plan to upload a large number of files, it's best to keep track locally of which files you already uploaded before uploading the next batch. You are ensured that an LFS file that is already committed will never be re-uploaded twice but checking it client-side can still save some time. This is what [`large_upload`] does for you.
+- **Use `hf_transfer`**: this is a Rust-based [library](https://github.com/huggingface/hf_transfer) meant to speed up uploads on machines with very high bandwidth. To use `hf_transfer`:
```

Comment:
Is `hf_transfer` still recommended when using the large upload feature?

Reply:
Yes it is! In that case, the number of workers pre-uploading files is limited to 1 since `hf_transfer` maxes out the CPU. This means the files will be hashed in parallel and then uploaded 1 by 1 which is a good behavior IMO.